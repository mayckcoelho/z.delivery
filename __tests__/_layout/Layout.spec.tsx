import 'jsdom-global/register'
import React from 'react'
import '../../config/setup'
import Layout from '../../src/containers/_layout'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { unmountComponentAtNode } from 'react-dom'

let container: HTMLDivElement | null = document.createElement('div')

beforeEach(() => {
  container && document.body.appendChild(container)
})

afterEach(() => {
  container && unmountComponentAtNode(container)
  container?.remove()
  container = null
})

describe('Layout should render correctly', () => {
  it('snapshot match', () => {
    const layout = shallow(<Layout />)

    expect(toJson(layout)).toMatchSnapshot()
  })

  it('should mount component', () => {
    mount(<Layout />)
  })
})
