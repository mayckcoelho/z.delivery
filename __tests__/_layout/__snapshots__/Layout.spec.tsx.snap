// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly snapshot match 1`] = `
<Fragment>
  <Header />
  <Footer />
</Fragment>
`;
